syntax = "proto3";

package ai_service;

message Message {
    string role = 1;
    string content = 2;
}

message ChatCompletionRequest {
    string model = 1;
    repeated Message messages = 2;
    string provider = 3;
    optional int32 max_tokens = 4;
    optional float temperature = 5;
}

message ChatCompletionResponse {
    string content = 1;
    string model = 2;
    string provider = 3;
}

service AIService {
    rpc GenerateCompletion (ChatCompletionRequest) returns (ChatCompletionResponse);
}